### DISCUSSION TITLE
How can I group updates to Python dependencies with their associated "types" packages?

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#24378] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Many Python packages don't ship type hints. Instead, typing is shipped as a separate package, conventionally named `types-foo` for a dependency `foo`; e.g. I use `paramiko` in one of my projects and I include `types-paramiko` as a dev dependency.

### What you tried so far.

I'm aware that I could manually create a group for each dependency that I know has a `types-` package but I want this to be generic, i.e. to apply to any such pair.

### Relevant debug logs

N/A

**Comment by rarkins** [comment, marked as answer]
There is no automated/generic way I can think of to achieve this today

**Reply by allanlewis** [reply]
Thanks, I suspected as much. I'll see if I can write a script to list all of the distinct `types-` packages across my monorepo and generate config snippets for Renovate as appropriate.

**Reply by rarkins** [reply]
*maybe* can be done using the "replace" handlebars helper we have for templating (replace types- in depName)
